### PR TITLE
CLN Apply style to Jupyter notebook

### PR DESCRIPTION
**Pull request by MarcoGorelli**
We apply many style checks to our source code files / Python snippets in documentation, so I've applied them to the (lone) Jupyter notebook.

Not adding anything to CI as it's arguably not worth it for just a single file

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Does this make a difference for anything?  I like uniformity as much as the next guy, but find the additional \\s make this marginally less readable.

**Comment by MarcoGorelli** [comment]
That's a good point...I've removed the extra spaces which `black` adds between function definitions and code, but I've kept the indentation / quoting changes

**Comment by jbrockmendel** [comment]
Hah, you must have read "\s" as a regex for whitespace, I was referring to literal backslashes.

**Comment by MarcoGorelli** [comment]
:rofl: indeed

Though doesn't this PR _remove_ the literal backslashes? e.g. in cell 9
```diff
- df.style.\
-     applymap(color_negative_red).\
-     apply(highlight_max)
+ df.style.applymap(color_negative_red).apply(highlight_max)
```

**Comment by jbrockmendel** [comment]
> Though doesn't this PR remove the literal backslashes? e.g. in cell 9

Sure.  More specifically I was looking at the single ' quotes that became \\"

I'd be fine with merging this just to clear the queue.  cc @jorisvandenbossche

**Comment by MarcoGorelli** [comment]
> Sure.  More specifically I was looking at the single ' quotes that became \\"

Ah yeah, they only _look_ like they've become like that because of how git shows the diff between notebooks.

E.g. git shows this diff:
```diff
-    "df = pd.DataFrame({'A': np.linspace(1, 10, 10)})\n",
-    "df = pd.concat([df, pd.DataFrame(np.random.randn(10, 4), columns=list('BCDE'))],\n",
-    "               axis=1)\n",
+    "df = pd.DataFrame({\"A\": np.linspace(1, 10, 10)})\n",
+    "df = pd.concat([df, pd.DataFrame(np.random.randn(10, 4), columns=list(\"BCDE\"))], axis=1)\n",
```

but if you open up the notebook in e.g. VSCode then it looks like this in this branch:

---
image
---

instead of this on master:

---
image
---

**Comment by MarcoGorelli** [comment]
closing as I don't think it's worth going back and forth for the sake of a single file, sorry for the noise 😄

**Comment by jorisvandenbossche** [comment]
BTW, I think this is perfectly reasonable to merge. The single vs double quote making it less readable in this diff is not really an issue IMO, that's just an artifact of the json format of the notebook, and doesn't impact the use of it (notebooks source are not really human-friendly to read anyway)

**Comment by MarcoGorelli** [comment]
:rofl: sure, reopening then

File: doc/source/user_guide/style.ipynb [hunk 1]
```diff
@@ -503,10 +500,11 @@
    "outputs": [],
    "source": [
     "# Compress the color range\n",
-    "(df.loc[:4]\n",
-    "    .style\n",
-    "    .background_gradient(cmap='viridis', low=.5, high=0)\n",
-    "    .highlight_null('red'))"
+    "(\n",
+    "    df.loc[:4]\n",
+    "    .style.background_gradient(cmap=\"viridis\", low=0.5, high=0)\n",
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/style.ipynb:505]: Is black also happy if you put the `.style` on the previous line?

File: doc/source/user_guide/style.ipynb [hunk 2]
```diff
@@ -854,10 +844,11 @@
...
    "outputs": [],
    "source": [
-    "(df.style\n",
-    "   .set_na_rep(\"FAIL\")\n",
-    "   .format(None, na_rep=\"PASS\", subset=[\"D\"])\n",
-    "   .highlight_null(\"yellow\"))"
+    "(\n",
+    "    df.style.set_na_rep(\"FAIL\")\n",
+    "    .format(None, na_rep=\"PASS\", subset=[\"D\"])\n",
+    "    .highlight_null(\"yellow\")\n",
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/style.ipynb:850]: We should maybe decide whether for those specific "method-chaining" cases we want to strictly follow black or not (it's quite typical for pandas, and the original layout might make it clearer what is going on)

File: doc/source/user_guide/style.ipynb [hunk 3]
```diff
@@ -991,14 +983,12 @@
...
     "bigdf = pd.DataFrame(np.random.randn(20, 25)).cumsum()\n",
     "\n",
-    "bigdf.style.background_gradient(cmap, axis=1)\\\n",
-    "    .set_properties(**{'max-width': '80px', 'font-size': '1pt'})\\\n",
-    "    .set_caption(\"Hover to magnify\")\\\n",
-    "    .set_precision(2)\\\n",
-    "    .set_table_styles(magnify())"
+    "bigdf.style.background_gradient(cmap, axis=1).set_properties(\n",
+    "    **{\"max-width\": \"80px\", \"font-size\": \"1pt\"}\n",
+    ").set_caption(\"Hover to magnify\").set_precision(2).set_table_styles(magnify())"
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/style.ipynb:991]: For example this one (related to my comment above), I don't directly find it an improvement ..

**Comment by jorisvandenbossche** [comment]
Actually looking at the code changes, there are a few examples in their of heavy method chaining that black doesn't always handle that nicely ... Not fully sure what to do with those.

**Comment by MarcoGorelli** [comment]
> Actually looking at the code changes, there are a few examples in their of heavy method chaining that black doesn't always handle that nicely 

True...OK, let's close then, not everything needs to follow `black` - sorry again for the noise, I'll focus on more pressing issues :smile:

**Pull request closed without merging**